dwl.c

```diff
@@ -1627,35 +1627,48 @@ rendermon(struct wl_listener *listener, void *data)
 	int skip = 0;
 	struct timespec now;
 
+
+	/* Let clients know a frame has been rendered */
+	clock_gettime(CLOCK_MONOTONIC, &now);
+
 	/* Render if no XDG clients have an outstanding resize and are visible on
 	 * this monitor.
 	 */
-	wl_list_for_each(c, &clients, link) {
-		if (m->un_map || c->surface.xwayland) {
+	if (m->un_map) {
+		wl_list_for_each(c, &clients, link) {
 			if (c->resize) {
 
 				/* Lie */
 				wlr_surface_send_frame_done(client_surface(c), &now);
 				skip = 1;
 			}
-			continue;
 		}
+	}
+	else {
+		wl_list_for_each(c, &clients, link) {
+			if (c->type != XDGShell) {
+				if (c->resize) {
```

Comment:
Only XDGShell clients have resize

Reply:
we need it regardless because later on we are accessing xdg surface stuff which will cause a crash if we try this with something that doesn't have an xdg surface. That's why the continue is there

Reply:
I meant, for xwayland clients `c->resize` is always zero, so this `if` is unneeded, you only need to keep the `continue`